test/layout/test_units.py

```diff
@@ -0,0 +1,57 @@
+from pathlib import Path
+
+import pytest
+
+from fpdf import FPDF
+from test.conftest import assert_pdf_equal
+
+HERE = Path(__file__).resolve().parent
+
+
+def test_unit_default(tmp_path):
+    """Test creating a PDF with no unit"""
+    pdf = FPDF()
+    pdf.set_font("Helvetica")
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="default")
+    assert_pdf_equal(pdf, HERE / "unit_default.pdf", tmp_path)
+
+
+def test_unit_adjustment(tmp_path):
+    """Test creating a PDF with no unit, then changing to another works"""
+    pdf = FPDF()
+    pdf.set_font("Helvetica")
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="default")
+    pdf.k = 1
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="pt")
+    assert_pdf_equal(pdf, HERE / "unit_changed.pdf", tmp_path)
+
+
+@pytest.mark.parametrize("unit", ["pt", "mm", "cm", "in"])
+def test_unit_str(tmp_path, unit):
+    """Test all of the unit strings"""
+    pdf = FPDF(unit=unit)
+    pdf.set_font("Helvetica")
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="{}".format(unit))
+    assert_pdf_equal(pdf, HERE / "unit_{}.pdf".format(unit), tmp_path)
+
+
+def test_unit_int(tmp_path):
+    """Test creating a PDF with an int unit"""
+    pdf = FPDF(unit=72)
+    pdf.set_font("Helvetica")
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="in")
+    assert_pdf_equal(pdf, HERE / "unit_in.pdf", tmp_path)
```

Comment:
Shouldn't the filename be `unit_int`?

Reply:
That's deliberate.  Added comments to explain what's going on.

It's not just testing creation, it's also making sure that one created with an `int` or `float` are exactly the same as one created with `unit="in"`.

If this were Python 2, then this test would fail, since k would end up causing integer division when configuring the margins.

Reply:
I've gone ahead and moved that to two different tests.